### PR TITLE
snap-bootstrap: test the new bind mounts

### PR DESCRIPTION
**Pull request by xnox**
Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

This incorporates https://github.com/snapcore/snapd/pull/10349 , then incorporates https://github.com/snapcore/snapd/pull/10350

The above pull requests can be merged first, or one can just merge this one. Note that 10350 tests must pass, and this one tests must pass to ensure we maintain compatible with old core-initrds.

### CODE REVIEW

File: cmd/snap-bootstrap/cmd_initramfs_mounts_test.go [hunk 1]
```diff
@@ -450,6 +453,27 @@ func ubuntuPartUUIDMount(partuuid string, mode string) systemdMount {
 	return mnt
 }
 
+// ubuntuPartUUIDMount returns a systemdMount for the partuuid disk, expecting
+// that the partuuid contains in it the expected label for easier coding
```
Comment (anonymouse64) [inline cmd/snap-bootstrap/cmd_initramfs_mounts_test.go:457]: this doc-comment is wrong

**Comment by anonymouse64** [comment]
this PR is fine, but I think for simplicity's sake it would make more sense to close this one, and incorporate the test changes here with those you made in #10350 so we can review the code alongside all of the test cases

**Comment by xnox** [comment]
> this PR is fine, but I think for simplicity's sake it would make more sense to close this one, and incorporate the test changes here with those you made in #10350 so we can review the code alongside all of the test cases

I am not too sure how to keep both types of tests. I.e. the old ones that assume that sysroot.mount is available; and these ones that are the future. I guess i have to copy some/any tests around and keep the sysroot.mount setup for them.

**Comment by anonymouse64** [comment]
> I guess i have to copy some/any tests around and keep the sysroot.mount setup for them.

yes that's exactly what I recommended in my review on that PR, let me know if you need help with that but I gave you some pointers for specific tests that should be sufficient coverage if you copy them to handle the case where the mount units exist

**Pull request closed without merging**